### PR TITLE
[Help Editor] Implemented Data Framework + Updated React 

### PR DESCRIPTION
**Pull request by HenriRabalais**
### Brief summary of changes
- Updated React Index file to current LORIS standard.
- Implemented the Data Framework for the Help Editor Module.

### To test this change...
- Make sure that everything is loading properly in the datatable
- Make sure the filter is filtering data properly

### CODE REVIEW

File: modules/help_editor/jsx/help_editor.js [hunk 1]
```diff
@@ -19,157 +19,118 @@ class HelpEditor extends React.Component {
...
    * Modify behaviour of specified column cells in the Data Table component
    *
    * @param {string} column - column name
    * @param {string} cell - cell content
-   * @param {array} rowData - array of cell contents for a specific row
-   * @param {array} rowHeaders - array of table headers (column names)
+   * @param {object} row - row content indexed by column
    *
-   * @return {*} a formatted table cell for a given column
+   * @return {*} a formated table cell for a given column
```
[the remark this answers is not in the document]
Comment (HenriRabalais) [reply modules/help_editor/jsx/help_editor.js]: yeah, probs a copy pasta

**Comment by HenriRabalais** [comment]
@johnsaigle Addressed.

File: modules/help_editor/test/help_editorTest.php [hunk 2]
```diff
@@ -134,6 +134,9 @@ function testHelpEditorWithoutPermission()
      */
     function testHelpEditorSearchByTopic()
     {
+         $this->markTestSkipped(
+             'Modifications needed to adapt to layout changes'
+         );
```
Comment (HenriRabalais) [inline modules/help_editor/test/help_editorTest.php]: @kongtiaowang This test was giving me trouble and I'm not sure how to resolve it.

File: modules/help_editor/test/help_editorTest.php [hunk 3]
```diff
@@ -153,6 +156,9 @@ function testHelpEditorSearchByTopic()
      */
     function testHelpEditorSearchByKeyword()
     {
+         $this->markTestSkipped(
+             'Modifications needed to adapt to layout changes'
+         );
```
Comment (HenriRabalais) [inline modules/help_editor/test/help_editorTest.php]: @kongtiaowang This test was giving me trouble and I'm not sure how to resolve it.

Comment (kongtiaowang) [reply modules/help_editor/test/help_editorTest.php]: @HenriRabalais OK, keep this mark here, I will fix this after this PR get merged.

**Comment by driusan** [comment]
Merging as per shen's comment above..

**Pull request merged**